packages/core/test/actions.test.ts

```diff
@@ -1446,6 +1447,67 @@ describe('sendParent', () => {
   });
 });
 
+describe('sendTo', () => {
+  it('should be able to send an event to an actor', (done) => {
+    const childMachine = createMachine<any, { type: 'EVENT' }>({
+      initial: 'waiting',
+      states: {
+        waiting: {
+          on: {
+            EVENT: {
+              actions: () => done()
+            }
+          }
+        }
+      }
+    });
+
+    const parentMachine = createMachine<{
+      child: ActorRefFrom<typeof childMachine>;
+    }>({
+      context: () => ({
+        child: spawn(childMachine)
+      }),
+      entry: sendTo((ctx) => ctx.child, { type: 'EVENT' })
+    });
+
+    interpret(parentMachine).start();
+  });
+
+  it('should be able to send an event from expression to an actor', (done) => {
+    const childMachine = createMachine<any, { type: 'EVENT'; count: number }>({
+      initial: 'waiting',
+      states: {
+        waiting: {
+          on: {
+            EVENT: {
+              cond: (_, e) => e.count === 42,
+              actions: () => done()
+            }
+          }
+        }
+      }
+    });
+
+    const parentMachine = createMachine<{
+      child: ActorRefFrom<typeof childMachine>;
+      count: number;
+    }>({
+      context: () => ({
+        child: spawn(childMachine),
+        count: 42
+      }),
+      entry: sendTo(
+        (ctx) => ctx.child,
+        // TODO: figure out how to avoid `as const`
```

Comment:
A simplified repro - [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggxsA9gJwEoQGYB4AqA+KAXigG8AoKSqAZwgDsATALigAoIA3FnASiII6IAlgzIBfMmVCQoAUQ71gAMWSIAtlhgFiMKBAAeweg2qwEKdNiF0MEZHIIB+OVBYBDOiEkMIcADZuyNAYAK50CEKIdDTGOIiaeobGpvBIaJhYHiB4eKxu5ukYLDAANFAA5hDA8oosrHyEBDV0yqoaWjwsgiKS0tDyRKRQfSwA5EoA8hOjANxQHtQA7nYs1MDI1uVQYlAAPkMjUKMAQjCos1BwiGHALHQhagBGdtvevgFBl1Fr83DFBZYsPI8JJaIw4nk4GV6vw2CRhuAIGNJtNtjweJIgA)

I'm not sure what to do about it, I've reached out to Devan. It's wild that autocomplete works at this position but yet the type is inferred as `string` 🤦

Reply:
@devanshj is a wizard. Based on his suggestions and some trial & error in our codebase and in the TS playground I got this: https://github.com/statelyai/xstate/pull/2847